nexus/types/src/deployment.rs

```diff
@@ -268,9 +379,64 @@ impl<'a> fmt::Display for BlueprintDisplay<'a> {
                 .unwrap_or_else(|| String::from("<none>"))
         )?;
 
-        writeln!(f, "\n{}", self.make_zone_table())?;
+        // Keep track of any sled_ids that have been seen in the first loop.
+        let mut seen_sleds = BTreeSet::new();
+
+        // Loop through all sleds that have physical disks and print a table of
+        // those physical disks.
+        //
+        // If there are corresponding zones, print those as well.
+        for (sled_id, disks) in &self.blueprint.blueprint_disks {
+            // Construct the disks subtable
+            let disks_table = BpSledSubtable::new(
+                BpPhysicalDisksSubtableSchema {},
+                disks.bp_generation(),
+                disks.rows(BpDiffState::Unchanged).collect(),
+            );
+
+            // Construct the zones subtable
+            let zones_table = match self.blueprint.blueprint_zones.get(sled_id)
+            {
+                Some(zones) => {
+                    let zones =
+                        BlueprintOrCollectionZonesConfig::from(zones.clone());
+                    BpSledSubtable::new(
+                        BpOmicronZonesSubtableSchema {},
+                        zones.bp_generation(),
+                        zones.rows(BpDiffState::Unchanged).collect(),
+                    )
+                    .to_string()
+                }
+                None => "".to_string(),
+            };
+            writeln!(
+                f,
+                "\n  sled: {sled_id}\n\n{disks_table}\n\n{zones_table}\n"
```

Comment:
Ah, this might be the source of all the blank space I saw above - maybe move this into the `match` and omit `\n\n{zones_table}` in the `None` case?

Reply:
Done!